ADCSensorReading.c

```
@@ -0,0 +1,46 @@
+#include "ADCSensorReading.h"
+
+int maxAnalogToDigitalConversion(int ADC_Resolution)
+{
+	return (pow(2,ADC_Resolution) - 2);
+}
+
+int stdAnalogToDigitalConvertedValue(int ADC_Resolution)
+{
+	return (pow(2,ADC_Resolution) - 1);
+}
+
+void currentSenseAtoDConversion(int ADCValues[], int numberOfSamples, int ADC_Resolution, int maxCurrentValue, int* currentConversionValues)
+{	
+	float currentValue, analogToDigitalCoversionValue;
+	
+	for (int i=0; i< numberOfSamples; i++)
+	{
+		analogToDigitalCoversionValue = (ADCValues[i]) / (maxAnalogToDigitalConversion(ADC_Resolution));
+		currentValue = (maxCurrentValue * analogToDigitalCoversionValue) / (maxAnalogToDigitalConversion(ADC_Resolution));
+		currentConversionValues[i] = round(currentValue);
+		if(currentConversionValues[i] < 0)
+			currentConversionValues[i] = abs(currentConversionValues[i]);
+	}	
+}
+
+void ignoreError(void)
+{

```

what is the use  of ignoreError function

---

actually i thought add ignore error condition in separate printf statement. yes i take this comment and update my code